api/processor_api.py

```diff
@@ -0,0 +1,41 @@
+import time
+from fastapi import FastAPI
+from fastapi.staticfiles import StaticFiles
+from fastapi.responses import FileResponse, StreamingResponse
+import uvicorn
+import os
+
+class ProcessorAPI:
+    """
+    The Webgui object implements a fastapi application and acts as an interface for users.
+    Once it is created it will act as a central application for viewing outputs.
+
+    :param config: Is a ConfigEngine instance which provides necessary parameters.
+    :param engine_instance:  A ConfigEngine object which store all of the config parameters. Access to any parameter
+        is possible by calling get_section_dict method.
+    """
+
+    def __init__(self, config):
+        self.config = config
+        self._host = self.config.get_section_dict("App")["Host"]
+        self._port = int(self.config.get_section_dict("App")["Port"])
+        self.app = self.create_fastapi_app()
+
+    def create_fastapi_app(self):
+        # Create and return a fastapi instance
+        app = FastAPI()
+
+        if os.environ.get('DEV_ALLOW_ALL_ORIGINS', False):
+            # This option allows React development server (which is served on another port, like 3000) to proxy requests
+            # to this server.
+            # WARNING: read this before enabling it in your environment:
+            # https://medium.com/@stestagg/stealing-secrets-from-developers-using-websockets-254f98d577a0
+            from fastapi.middleware.cors import CORSMiddleware
+            app.add_middleware(CORSMiddleware, allow_origins='*', allow_credentials=True, allow_methods=['*'],
+                               allow_headers=['*'])
+        app.mount("/static", StaticFiles(directory="/repo/data/web_gui/static"), name="static")
```

Comment:
Does "web_gui/static" comes from the frontend files? Sorry I have no clues on how these work together, just thought maybe it has not to be there if it is for the dashboard. Correct me if I am wrong

Reply:
maybe I am confused as the old name "web_gui" was used for front-end.

Reply:
Never mind, I think I figured it out and asked my question in "ui/web_gui.py" changes where you have removed this line from.

Reply:
This basically allows the access to the video streams. Renaming it sounds correct